reframe/frontend/executors/__init__.py

```diff
@@ -386,20 +404,38 @@ def runall(self, testcases, restored_cases=None):
             if self._max_retries:
                 restored_cases = restored_cases or []
                 self._retry_failed(testcases + restored_cases)
-
+        except MaxFailError as e:
+            abort_reason = e
+        except KeyboardInterrupt:
+            abort_reason = "keyboard interrupt"
+            raise
+        except ReframeForceExitError:
+            abort_reason = "reframe error"
+            raise
         finally:
             # Print the summary line
-            num_failures = len(self._stats.failures())
+            num_failures = len(self._stats.failed())
+            num_completed = len(self._stats.completed())
+            if abort_reason or num_failures:
```

Comment:
I don't understand this `if`. I'm trying to simplify the exception handling and I have a design that wouldn't even the `except` above nor the `abort_reason`. So I'm wondering why not simply `if num_failures` here?

Reply:
I removed completely the `abort_reason`. Let me know if you see something that I can't figure out.

Reply:
No, I think it is okay.